api/reference/configuration_script_payloads.md

```diff
@@ -0,0 +1,67 @@
+---
+---
+
+## Configuration Script Payloads
+
+Management of configuration script payloads is provided through the following
+collection:
+
+``` data
+/api/configuration_script_payloads
+```
+
+The following actions on configuration script payloads are available:
+
+- [Querying Configuration Script Payloads](#querying-configuration-script-payloads)
+
+- [Editing Configuration Script Payloads](#editing-configuration-script-payloads)
+
+- [Updating Configuration Script Payloads](#updating-configuration-script-payloads)
+
+- [Querying Configuration Script Payload Authentication](#querying-configuration-script-payload-authentication)
+
+- [Creating Configuration Script Payload Authentication](#creating-configuration-script-payload-authentication)
+
+### Querying Configuration Script Payloads
+
+Query all configuration script payloads in the system:
+
+    GET /api/configuration_script_payloads
+
+Get the details on a specific configuration script payload:
+
+    GET /api/configuration_script_payloads/:id
+
+### Editing configuration script payloads
+
+Configuration script payloads can be edited by using a POST to the configuration script payloads collection:
+
+    POST /api/configuration_script_payloads
+
+### Updating configuration script payloads
+
+Configuration script payloads can be updated by using a PUT method or PATCH method.
+
+PUT method:
+
+    PUT /api/configuration_script_payloads/:id
+
+Patch method:
+
+    PATCH /api/configuration_script_payloads/:id
```

Comment:
`PATCH /api/configuration_script_payloads/1 {"description": "A Helpful Description"}`

```json
{
  "results": [
    {
      "href": "http://localhost:3000/api/configuration_script_payloads/1",
      "description": "A Helpful Description",
      "credentials": {
        "api_user.$": {
          "credential_ref": "manageiq_api",
          "credential_field": "userid"
        },
        "api_password.$": {
          "credential_ref": "manageiq_api",
          "credential_field": "password"
        }
      },
      "id": "1",
      "manager_id": "2",
      "manager_ref": null,
      "name": "provision-vm-service/list-providers.asl",
      "variables": null,
      "created_at": "2023-11-08T17:43:30Z",
      "updated_at": "2023-11-15T20:14:28Z",
      "survey_spec": null,
      "inventory_root_group_id": null,
      "type": "ManageIQ::Providers::Workflows::AutomationManager::Workflow",
      "parent_id": null,
      "configuration_script_source_id": "2",
      "run_by_userid": null,
      "payload": "{\n  \"Comment\": \"List Providers.\",\n  \"StartAt\": \"ListProviders\",\n  \"States\": {\n    \"ListProviders\": {\n      \"Type\": \"Task\",\n      \"Resource\": \"docker://docker.io/agrare/list-providers:latest\",\n      \"End\": true,\n      \"Credentials\": {\n        \"api_user.$\": \"$.api_user\",\n        \"api_password.$\": \"$.api_password\",\n        \"api_token.$\": \"$.api_token\",\n        \"api_bearer_token.$\": \"$.api_bearer_token\"\n      },\n      \"Parameters\": {\n        \"API_URL.$\": \"$$.Execution._manageiq_api_url\",\n        \"VERIFY_SSL.$\": \"$.dialog.dialog_verify_ssl\",\n\n        \"PROVIDER_TYPE\": \"ManageIQ::Providers::Vmware::InfraManager\"\n      }\n    }\n  }\n}\n",
      "payload_type": "json",
      "context": null,
      "output": null,
      "status": null,
      "miq_task_id": null,
      "payload_valid": true,
      "payload_error": null
    }
  ]
}
```

Reply:
This example is added.